src/bootstrap/src/core/download.rs

```diff
@@ -291,7 +291,7 @@ impl Config {
             let mut short_path = t!(original_path.strip_prefix(directory_prefix));
             let is_builder_config = short_path.to_str() == Some(BUILDER_CONFIG_FILENAME);
 
-            if !short_path.starts_with(pattern) && (is_ci_rustc && !is_builder_config) {
+            if !(short_path.starts_with(pattern) || is_ci_rustc && is_builder_config) {
```

Comment:
Could you please add parentheses like this:
```rust
short_path.starts_with(pattern) || (is_ci_rustc && is_builder_config)
```
The current logic is correct, but it took me a while to realize that :laughing: I think that this makes it immediately clear what's going on.